src/org/zaproxy/zap/extension/browserView/BrowserPanel.java

```diff
@@ -88,4 +159,44 @@ private static String toURL(String str) {
             return null;
         }
     }
+    
+    private int getWebsiteHeight(){
+        String script =   "var body = document.body, html = document.documentElement;"
+                        + "Math.max(body.offsetHeight, html.offsetHeight);";
+        return Integer.parseInt(engine.executeScript(script).toString());
+    }
+    
+    private void adjustPanelHeightToWebsite(){
+        final AtomicReference<Integer> webSiteHight = new AtomicReference<>();
+        final CountDownLatch latch = new CountDownLatch(1);
+        
+        Platform.runLater(new Runnable() {
+             @Override 
+             public void run() {
+                 int height = getWebsiteHeight();
+                 webSiteHight.set(height);
+                 latch.countDown();
+             }
+         });
+        
+        try
+        {
+            latch.await(3, TimeUnit.SECONDS);
+        }catch(Exception ex){
+            logger.error("Error while waiting for determining websiteHeight in JavaFX-Thread.", ex);
```

Comment:
Should probably warn or even debug, there's not much that can be done with this info (we don't get what interrupted the EDT).